common/src/main/java/org/opensearch/ml/common/transport/model_group/MLRegisterModelGroupInput.java

```diff
@@ -17,39 +17,46 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
 
 @Data
 public class MLRegisterModelGroupInput implements ToXContentObject, Writeable{
 
     public static final String NAME_FIELD = "name"; //mandatory
-    public static final String DESCRIPTION_FIELD = "description";
-    public static final String TAGS_FIELD = "tags"; //mandatory
-    public static final String MODEL_IDS_FIELD = "model_ids";
+    public static final String DESCRIPTION_FIELD = "description"; //optional
+    public static final String TAGS_FIELD = "tags"; //optional
+    public static final String BACKEND_ROLES_FIELD = "backend_roles"; //optional
+    public static final String IS_PUBLIC = "is_public"; //optional
+    public static final String ADD_ALL_BACKEND_ROLES = "add_all_backend_roles"; //optional
 
     private String name;
     private String description;
-    private List<String> tags;
-    private List<String> models;
+    private Map<String, Object> tags;
+    private List<String> backendRoles;
+    private Boolean isPublic;
+    private Boolean addAllBackendRoles;
```

Comment:
[nit] Using "is" or "has" as prefixes for boolean variables can help improve code readability by making it clear that the variable represents a condition or state. May be we can rename this variable?

Reply:
something like isAddAllBackendRoles? When user specifies this field in the request, maybe addAllBackendRoles can be easier for user to know its purpose

Reply:
Customer facing variable/parameter and internal variable doesn't need to be the same name. if the addAllBackendRoles represents a flag indicating whether backend roles should be added, you could rename it to `shouldAddBackendRoles`.

Reply:
If a model has all backend roles, doesn't that mean it's a public model, is this `addAllBackendRoles` duplicated with `isPublic`?